scripts/exregional_process_lightning.sh

```diff
@@ -0,0 +1,207 @@
+# (scrfunc_fp), the name of that file (scrfunc_fn), and the directory in
+# which the file is located (scrfunc_dir).
+#
+#-----------------------------------------------------------------------
+#
+scrfunc_fp=$( readlink -f "${BASH_SOURCE[0]}" )
+scrfunc_fn=$( basename "${scrfunc_fp}" )
+scrfunc_dir=$( dirname "${scrfunc_fp}" )
+#
+#-----------------------------------------------------------------------
+#
+# Print message indicating entry into script.
+#
+#-----------------------------------------------------------------------
+#
+print_info_msg "
+========================================================================
+Entering script:  \"${scrfunc_fn}\"
+In directory:     \"${scrfunc_dir}\"
+
+This is the ex-script for the task that runs lightning preprocess
+with FV3 for the specified cycle.
+========================================================================"
+#
+#-----------------------------------------------------------------------
+#
+# Extract from CDATE the starting year, month, day, and hour of the
+# forecast.  These are needed below for various operations.
+#
+#-----------------------------------------------------------------------
+#
+START_DATE=$(echo "${CDATE}" | sed 's/\([[:digit:]]\{2\}\)$/ \1/')
+YYYYMMDDHH=$(date +%Y%m%d%H -d "${START_DATE}")
+JJJ=$(date +%j -d "${START_DATE}")
+
+YYYY=${YYYYMMDDHH:0:4}
+MM=${YYYYMMDDHH:4:2}
+DD=${YYYYMMDDHH:6:2}
+HH=${YYYYMMDDHH:8:2}
+YYYYMMDD=${YYYYMMDDHH:0:8}
+
+YYJJJHH=$(date +"%y%j%H" -d "${START_DATE}")
+PREYYJJJHH=$(date +"%y%j%H" -d "${START_DATE} 1 hours ago")
+
+#
+#-----------------------------------------------------------------------
+#
+# Get into working directory
+#
+#-----------------------------------------------------------------------
+#
+print_info_msg "$VERBOSE" "
+Getting into working directory for lightning process ..."
+
+cd_vrfy ${DATA}
+
+fixgriddir=$FIXgsi/${PREDEF_GRID_NAME}
+
+print_info_msg "$VERBOSE" "fixgriddir is $fixgriddir"
+
+#
+#-----------------------------------------------------------------------
+#
+# link or copy background and grid files
+#
+#-----------------------------------------------------------------------
+
+cp_vrfy ${fixgriddir}/fv3_grid_spec          fv3sar_grid_spec.nc
+
+#-----------------------------------------------------------------------
+#
+# Link to the NLDN data
+#
+#-----------------------------------------------------------------------
+run_lightning=false
+filenum=0
+LIGHTNING_FILE=${LIGHTNING_ROOT}/vaisala/netcdf
+for n in 00 05 ; do
+  filename=${LIGHTNING_FILE}/${YYJJJHH}${n}0005r
+  if [ -r ${filename} ]; then
+  ((filenum += 1 ))
+    ln -sf ${filename} ./NLDN_lightning_${filenum}
+  else
+   echo " ${filename} does not exist"
+  fi
+done
+for n in 55 50 45 40 35 ; do
+  filename=${LIGHTNING_FILE}/${PREYYJJJHH}${n}0005r
+  if [ -r ${filename} ]; then
+  ((filenum += 1 ))
+    ln -sf ${filename} ./NLDN_lightning_${filenum}
+    run_lightning=true
+  else
+   echo " ${filename} does not exist"
+  fi
+done
```

Comment:
This is all essentially the same logic, just stepping from 25 minutes before the hour to 5 mins after....I'm curious if it might lend itself better to a single over time?

```
for incr in $(seq -25 5 5) ; do 
  filedate=$(date +"%y%j%H%M" -d "${START_DATE} ${incr} minutes ")
  filename=${LIGHTNING_FILE}/${filedate}0005r
  ...
done

```


With this approach, that might mean a significant cleanup of the date variables needed earlier in this script.